商品出品機能-JS編

ymrt43 opened the pull request:
# What
以下を主に_itemForm.html.haml/market.scss/item.js/category-form.js作成・編集
- 表記auction ->marketに変更(クラス名、ファイル名リネームしたので注意)
- new.html.hamlを3つの部分テンプレートに分岐 -> edit.html.haml反映
(_itemForm/_marketHeader/_marketFooter)
- 画像プレビュー表示
- 画像複数枚選択
- 画像削除 ※edit.html.hamlでエラー出ています！
- 出品機能のエラーハンドリング
- Item.rbにバリデーション追加
- text_area文字数表示
- カテゴリー選択ツリー連動表示
- 商品価格の上下限度額表示
- 商品内容編集ページ作成
- 出品成功のリダイレクト先変更
- itemsテーブルの:user_id -> :seller_idに変更
- items/users/categoriesテーブルのマイグレーションファイル 結合/削除

# Why
商品出品機能を単体テスト以外完了させたので、
開発期間を鑑み一先ず共有するため

ghost commented:
Sider detected **1** warning on analyzing the commit ed3cec3.

- ⚠️ [Querly](https://sider.review/gh/repos/291402631/pulls/19/logs/aea17d07-3066-4a21-a0ce-898e848118c5)

We recommend fixing them as possible by updating the dependencies, configuring the analysis tool, [configuring `sider.yml`](https://sider.review/gh/repos/291402631/pulls/19#config), [turning off unused tools](https://sider.review/gh/repos/291402631/settings), and so on.

If you have problems or questions still, feel free to ask us via chat. 💬

---

You can [turn off this notification](https://sider.review/gh/repos/291402631/settings#notifications) if you don't want to receive it from now on.

ymrt43 committed:
gem 'jquery-rails' install

ymrt43 committed:
Create assets/javascripts/application.js

ymrt43 committed:
Update in application.html.haml to read application.js

ymrt43 committed:
Update manifest.js to read application.js

ymrt43 committed:
Update application.js to read all of js files under its tree

ymrt43 committed:
Create item.js

ymrt43 committed:
Merge branch 'master' into JS_function_item_up_for_sale

ymrt43 committed:
Merge branch 'master' into JS_function_item_up_for_sale

ymrt43 committed:
change column name in new.html.haml

area to prefecture_id

ymrt43 committed:
Delete validation of prefecture presence

ymrt43 committed:
Put custom data (data-index) at .FormBlock__iSelection

ymrt43 committed:
Variable definition 'fileIndex' in item.js

ymrt43 committed:
Add to call buildFileField with fileIndex

ymrt43 committed:
Add event when click remove-image-btn

ymrt43 committed:
Add remove element at images field in items/new.html.haml

ymrt43 committed:
Add function buildFileField in item.js

ymrt43 committed:
Change "auction"or"Auction" in class-names & file-name into "market"or"Market"

misunderstandings

ymrt43 committed:
Delete 'e' of click event in item.js

ymrt43 committed:
divide items/index.html.haml into 3 partial files

ymrt43 committed:
Change to use partial template in buy.html.haml

ymrt43 committed:
file change

ymrt43 committed:
Merge branch 'master' into JS_function_item_up_for_sale

ymrt43 committed:
Change action for :set_item

ymrt43 committed:
Change DOM for images form in items/new.html.haml

ymrt43 committed:
Replace where redirect to when item successed to update

ymrt43 committed:
Update motions in item.js to select file of images

ymrt43 committed:
Add before_action :set_parents in items_controller.rb

ymrt43 committed:
fix position of place to show previews

ymrt43 committed:
change first selects of category_form @categories -> @parents

ymrt43 committed:
Add #search in items_controller.rb

ymrt43 committed:
Add routes of items#search

ymrt43 committed:
Set @children in format.json :items#search

ymrt43 committed:
Create items/search.json.jbuilder

ymrt43 committed:
Create category-form.js

ymrt43 committed:
Delete duplicate css

ymrt43 committed:
Add :search into except_action of set_item in items_controller.rb

ymrt43 committed:
Inclease style type of select_category_form

ymrt43 committed:
Add setting of @grandchildren in items#search

ymrt43 committed:
Add json.array @grandchildren

ymrt43 committed:
Add ID & Change class name to show category_select_form

ymrt43 committed:
Create ajax for select box of category

ymrt43 committed:
Recreate DB

ymrt43 committed:
Change column name about association with User in item.rb

ymrt43 committed:
Change column name of association with Item in user.rb

ymrt43 committed:
Change method in order to change column name in items

ymrt43 committed:
Replace where redirect when items#create & Change column_name in item_params

ymrt43 committed:
Add text_conter_function

ymrt43 committed:
Add style for item_form:focus

ymrt43 committed:
Add calculator item.js & etc

etc meens

add event into counter of description's text

class names to remove( or so) change

ymrt43 committed:
Set @prefectures, delete @categories

ymrt43 committed:
Add validations of Item

ymrt43 committed:
Fix search default in routes.rb

ymrt43 committed:
Change to solve errors of item_up_for_sale function

ymrt43 committed:
add gem 'rspec-rails'

ymrt43 committed:
Generate files for rspec

ymrt43 committed:
Create spec/models/item_spec.rb

ymrt43 committed:
gem 'factory_bot_rails' install

akyhysa commented:
LGTM!!
コードだけを見ると私の理解不足でわからない箇所もありますが、
挙動に問題が無い様であれば、メンターレビューを受けてマージ宜しくお願いします。
"itemsテーブルの:user_id -> :seller_idに変更"で実装済みの購入機能の変更を行います。

ymrt43 committed:
Change way of moving imageBox in items#new

TOMARUMARU commented on app/controllers/items_controller.rb:
```diff
@@ -1,38 +1,40 @@
     @item = Item.new(item_params)
     if @item.save
-      redirect_to root_path
+      redirect_to "/items/#{@item.id}"
```
pathはprefixを使いましょう。

ymrt43 replied:
修正しました！

TOMARUMARU commented on app/controllers/items_controller.rb:
```diff
@@ -1,38 +1,40 @@
+  
+  def update
+    if @item.update(item_params)
+      redirect_to "/items/#{@item.id}"
```
pathはprefixを使いましょう。

ymrt43 replied:
修正しました！

ymrt43 committed:
Change url -> prefix in items_controller.rb

TOMARUMARU reviewed (commented):
LGTMです。お疲れ様でした。mergeして次に進みましょう。
